### PR TITLE
Add `ParetoNBDModel`

### PR DESCRIPTION
**Pull request by ColtAllen**
This PR closes https://github.com/pymc-labs/pymc-marketing/issues/127 and is a copy of https://github.com/pymc-labs/pymc-marketing/pull/177, which had to be closed due to merge conflicts. This is a backup branched off my previous working branch, so if there are still issues with merging, let me know and I'll close this PR, manually copy the changes into a new branch off `main`, and recreate.

This model is still in experimental status pending https://github.com/pymc-labs/pymc-marketing/pull/298.

----
:books: Documentation preview :books:: https://pymc-marketing--307.org.readthedocs.build/en/307/

### CODE REVIEW

File: tests/clv/models/test_pareto_nbd.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        "fit_method, rtol",
+        [
+            ("mcmc", 0.1),
+            ("map", 0.2),
+        ],
+    )
+    def test_model_convergence(self, fit_method, rtol):
+        # Edit priors here for convergence testing
+        # Note that None/pm.HalfFlat is extremely slow to converge
+        r_prior = pm.Weibull.dist(alpha=2, beta=1)
```
Comment (ricardoV94) [inline tests/clv/models/test_pareto_nbd.py:132]: Aren't these the defaults now?

**Review by ricardoV94 — COMMENTED** [review]
Looks good, just one also question. I think we should merge this after the ModelBuilder, maybe we can do both today

**Comment by ricardoV94** [comment]
Hmm, the slow tests are not running on the CI, that's why the codecov was complaining...

**Comment by ColtAllen** [comment]
I've been working in the wrong backup branch for the `ModelBuilder` integration 🐥 Closing & reopening yet again

**Pull request closed without merging**